R/dash.R

```diff
@@ -1439,8 +1461,14 @@ Dash <- R6::R6Class(
       # insert meta tags if present
       meta_tags <- all_tags[["meta_tags"]]
 
-      private$.index <- sprintf(
-        '<!DOCTYPE html>
+      if (!is.null(private$custom_index)) {
+        interpolated_index <- glue::glue(private$custom_index)
+        
+        private$.index <- interpolated_index
```

Comment:
One thing I don't think I see here that we likely want to implement are the interpolation keys -- see the comments in this merged PR for more info:

https://github.com/plotly/dash/pull/286

Let me know if I can help with anything here, but the general goal is to make the implementation as similar as we can to what currently exists on the Python side (within reason!).

Reply:
Implemented in 036e4e. It's not as clean as I'd want it to be, but I'd be interested to hear your thoughts on it.

Reply:
I've tried to make the implementation a little more concise while aiming for 🌴 code, in 5ae7549 and 39d52e0.